libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -448,6 +448,39 @@ int AP_HAL__I2CDevice_read_registers(lua_State *L) {
     return success;
 }
 
+int AP_HAL__I2CDevice_write_registers(lua_State *L) {
+    const int args = lua_gettop(L);
+    bool multi_byte;
+    if (args == 2) {
+        multi_byte = false;
+    } else if (args == 3) {
+        multi_byte = true;
+    } else {
+        return luaL_argerror(L, args, "expected 1 or 2 arguments");
+    }
+
+    AP_HAL::I2CDevice * ud = *check_AP_HAL__I2CDevice(L, 1);
+
+    const uint8_t addr = get_uint8_t(L, 2);
+
+    std::vector<uint8_t> values;
+    values.push_back(addr);
```

Comment:
No allowed std library's, sorry. 

You will have to allocate a buffer using,

`char *buffer = (char*)luaM_malloc(L, len);`

So you need to grab the length of the table before hand. Then loop over the table and fill in the buffer. 

Also means you can't use the `get_uint8_t` helper is it will never return if it fails you need to free the buffer before  you return the error.

Reply:
Ah, I didn't know about the `std` library, sorry!

I wanted to do it "properly" with the array, but I don't fully understand how this mapping from Lua to C works and how to get the length of the table. 😓 
Would `lua_rawlen(L, 3)` work?

Reply:
@IamPete1, is this any better now?